job-dsl-plugin/src/main/groovy/javaposse/jobdsl/plugin/ExecuteDslScripts.java

```diff
@@ -272,8 +286,10 @@ public boolean perform(final AbstractBuild<?, ?> build, final Launcher launcher,
             Collection<SeedReference> seedJobReferences = descriptor.getTemplateJobMap().get(templateName);
             Collection<SeedReference> matching = Collections2.filter(seedJobReferences, new SeedNamePredicate(seedJobName));
             if (!matching.isEmpty()) {
-                seedJobReferences.removeAll(matching);
```

Comment:
It would be better to just skip both for loops. As far as I see, in case of a dry run these loops have no effect.